plugin/bufferline.vim

```diff
@@ -96,16 +95,32 @@ let bufferline = extend({
 \ 'letters': 'asdfjkl;ghnmxcbziowerutyqpASDFJKLGHNMXCBZIOWERUTYQP',
 \}, get(g:, 'bufferline', {}))
 
-" Default icons
-let icons = extend({
-\ 'bufferline_default_file': '',
-\ 'bufferline_separator_active':   '▎',
-\ 'bufferline_separator_inactive': '▎',
-\ 'bufferline_close_tab': '',
-\ 'bufferline_close_tab_modified': '●',
-\ 'bufferline_arrow_left': '',
-\ 'bufferline_arrow_right': '',
-\}, get(g:, 'icons', {})) " 
+function! bufferline#init_icons() abort
+  " Older bufferline.icons = v:true/v:false configurations
+  if type(g:bufferline.icons) == v:t_bool
+    let g:bufferline.icons = {'devicons': g:bufferline.icons}
+  " Older bufferline.icons = 'numbers' configurations
+  elseif type(g:bufferline.icons) == v:t_string
+    let g:bufferline.icons = {'numbers': v:true}
+  endif
+
+  " Default icons
+  let g:bufferline.icons = extend({
+  \ 'bufferline_default_file': '',
+  \ 'bufferline_separator_active':   '▎',
+  \ 'bufferline_separator_inactive': '▎',
+  \ 'bufferline_close_tab': '',
+  \ 'bufferline_close_tab_modified': '●',
+  \ 'bufferline_arrow_left': '',
+  \ 'bufferline_arrow_right': '',
+  \ 'devicons': v:true,
+  \ 'numbers': v:false,
+  \}, g:bufferline.icons)
+
+  return g:bufferline.icons
```

Comment:
Unused return value in the initial `call`, but it is used to shorten the syntax up in the `render` code.